packages/xr/src/space.ts

```diff
@@ -73,14 +76,21 @@ function computeOriginReferenceSpaceOffset(object: Object3D, origin: Object3D |
 
 function getXRSpaceFromAncestorsRec(
   object: Object3D | null,
+  targetObject: Object3D,
   targetOffsetMatrix: Matrix4 | undefined,
 ): XRSpace | undefined {
   if (object == null) {
     return undefined
   }
   if (object.xrSpace != null) {
+    // Calculate offset using world matrices instead of accumulating local matrices.
+    // This correctly handles components that override updateWorldMatrix.
+    if (targetOffsetMatrix != null) {
+      object.updateWorldMatrix(true, false)
```

Comment:
this line is not necassary since object.updateWorldMatrix(true, false) is already executed in line 54 and since the object here is always an ancestor of the object in line 54 (please remove this line but maybe state in the requirements of getXRSpaceFromAncestorsRec, that the world matrices of "object" and all its ancestors and targetObject must be up to date

Reply:
Thanks.